android/library/src/main/java/io/logto/android/Logto.kt

```diff
@@ -2,23 +2,52 @@ package io.logto.android
 
 import android.app.Application
 import android.content.Context
-import io.logto.android.api.LogtoService
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.android.Android
+import io.ktor.client.features.logging.Logging
+import io.logto.client.service.LogtoService
 import io.logto.android.auth.AuthManager
 import io.logto.android.auth.browser.BrowserSignInFlow
 import io.logto.android.auth.browser.BrowserSignOutFlow
-import io.logto.android.client.LogtoApiClient
-import io.logto.android.config.LogtoConfig
-import io.logto.android.exception.LogtoException
-import io.logto.android.model.TokenSet
+import io.logto.android.client.LogtoAndroidClient
+import io.logto.client.config.LogtoConfig
+import io.logto.client.exception.LogtoException
+import io.logto.client.model.TokenSet
 import io.logto.android.storage.TokenSetStorage
 import org.jose4j.jwt.JwtClaims
 
 class Logto(
-    private val logtoConfig: LogtoConfig,
+    logtoConfig: LogtoConfig,
     application: Application,
     useStorage: Boolean = true,
 ) {
-    fun isAuthenticated(): Boolean = tokenSet != null
+    private companion object {
+        private const val STORAGE_SHAREDPREFERENCES_NAME_PREFIX = "io.logto.android"
+    }
+
+    private val logtoService = LogtoService(HttpClient(Android) {
```

Comment:
just wondering: seems like `logtoService` is only consumed by the `logtoClient`.  Would it better to create the `logtoService` within the `logtoClient`?